src/OpenTelemetry.ResourceDetectors.Container/ContainerResourceDetector.cs

```diff
@@ -18,6 +21,13 @@ public class ContainerResourceDetector : IResourceDetector
     private const string Filepath = "/proc/self/cgroup";
     private const string FilepathV2 = "/proc/self/mountinfo";
     private const string Hostname = "hostname";
+    private const string K8sServiceHostKey = "KUBERNETES_SERVICE_HOST";
+    private const string K8sServicePortKey = "KUBERNETES_SERVICE_PORT_HTTPS";
+    private const string K8sNamespaceKey = "KUBERNETES_NAMESPACE";
```

Comment:
Namespace should be published in `/var/run/secrets/kubernetes.io/serviceaccount/namespace`. Probably the best way to read it from file instead of asking user to configure passing it in environment variable though downwards api.